### ISSUE TITLE
References with square brackets in the name cannot be located

### ISSUE THREAD

**Issue by daveshanley** [pb33f/libopenapi#112]
`#/components/schemas/ResourceList[InsuranceResource]` will cause a failure when looking up the reference. Need to add support for this type of name format.

Although, it's not standard, it is still technically valid.

**Comment by WillGunn** [comment]
I dug into this a bit in the bug report mentioned, for posterity:
Hmm, we are using a custom TypeName filter in Swashbuckle.Aspnetcore,
```c#
public static string GetTypeName(Type type)
        {
            if (type.IsGenericType)
            {
                var paramTypeList = new List<string>();
                foreach (var genericParam in type.GenericTypeArguments)
                {
                    paramTypeList.Add(GetTypeName(genericParam));
                }

                var genericType = string.Join(",", paramTypeList);
                var name = type.Name.Substring(0, type.Name.IndexOf("`"));

                return $"{name}[{genericType}]";
            }
            else
            {
                return type.Name;
            }
        }
``` 
we used this code back with swagger/openapi 2.0, and looking at the spec, there were no key limitations specified from what I can tell, so the addition of that was a breaking change from 2.0 to 3.0, but it seems most tools don't impose that 3.0 limitation.

It doesn't cause issues in the rest of our tools, but I do see that pasting the file into https://editor-next.swagger.io/ complains about the [] symbols in the component keys (regex ^[a-zA-Z0-9\.\-_]+$ ).

Because of this key string limitation, there isn't really a good way to represent generics using only alphanumeric and .-_

**Comment by daveshanley** [comment]
It is fixed in `v0.9.0`
